### DISCUSSION TITLE
Navigating Steps doesn't load new data

### DISCUSSION THREAD

**Discussion by khalidmaquilang** [spatie/laravel-livewire-wizard#113] [Q&A]
When navigating it retains the old data.. example in my application it stores all the data in the page when pressing next so after you press next and then go back to the previous page by using ``$this->previousStep()`` it doesn't not run ``mount``, it retains the old data(data before storing)

**Comment by seche** [comment]
> When navigating it retains the old data.. example in my application it stores all the data in the page when pressing next so after you press next and then go back to the previous page by using `$this->previousStep()` it doesn't not run `mount`, it retains the old data(data before storing)

I noticed that as well. I had to use `$this->state()->forStep($step->stepName)` inside navigation.blade to pull the "new" data array rather than the "old" data. 

For example if the label changed, `$step->label` (old) `$this->state()->forStep($step->stepName)['label']` (new)
